addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3566,6 +3566,30 @@ registry.BackgroundImage = SnippetOptionWidget.extend({
             this.$target.trigger('background_changed', [previewMode]);
         }
     },
+    /**
+     * Changes the main color of dynamic SVGs.
+     *
+     * @see this.selectClass for parameters
+     */
+    async dynamicColor(previewMode, widgetValue, params) {
+        const currentSrc = getBgImageURL(this.$target[0]);
+        switch (previewMode) {
+            case true:
+                this.previousSrc = currentSrc;
+                break;
+            case 'reset':
+                this.$target.css('background-image', `url('${this.previousSrc}')`);
+                return;
+        }
+        const newURL = new URL(currentSrc, window.location.origin);
```

Comment:
quid about cdn ? lazyloading? ...

Reply:
Not sure what you mean here. Dynamic SVGs are always local.

Reply:
pq ca ne pourrait pas utiliser le cdn ?

Reply:
We need to replace the colors inside the SVG. If we do end up using a CDN that also replaces the colors in the SVG then this option would need to take that into account in its computeWidgetVisibility method but I don't see why this line would be a problem.